data/campaigns/Secrets_of_the_Ancients/utils/zombie_recruit_dialog.lua

```diff
@@ -87,12 +87,18 @@ wml.variables["recruitedZombieType"] = "cancel" -- default value
 if zExists==false then
     gui.show_prompt("", _ "There are no corpses available.", "")
 else
-    --TODO: this seems not to be synced.
-    local returned = gui.show_dialog(wml.get_child(zombie_recruit_dialog, 'resolution'), preshow, postshow)
-    if  returned ~= -2 and sides[1].gold  < recruitCost then
-        gui.show_prompt("", _ "You do not have enough gold to recruit that unit", "")
-    elseif returned ~= -2 and (sides[1].gold ) >= recruitCost then
-        wml.variables["recruitedZombieType"] = recruitedType
-        wml.variables["recruitedZombieCost"] = recruitCost
-    end
+    local result = wesnoth.sync.evaluate_single(
+        function()
+            local res = gui.show_dialog(wml.get_child(zombie_recruit_dialog, 'resolution'), preshow, postshow)
+            if res == -2 then
+                return {recruitCost = 0, recruitedType = "cancel"}
+            end
+            if sides[1].gold  < recruitCost then
+                gui.show_prompt("", _ "You do not have enough gold to recruit that unit", "")
+                return {recruitCost = 0, recruitedType = "cancel"}
+            end
+            return {recruitCost = recruitCost, recruitedType = recruitedType}
+        end)
```

Comment:
This indentation _feels_ unbalanced. I'd probably prefer to move `function()` to line 90 and push everything back one level. Alternatively, maybe moving the closing parenthesis to its own line?

Reply:
The latest push does this whitespace change (only).